core/src/mast/node/mod.rs

```diff
@@ -33,9 +34,49 @@ use crate::{
     mast::{MastForest, MastNodeId, Remapping},
 };
 
+#[enum_dispatch]
+pub trait MastNodeTrait {
```

Comment:
We should merge the existing `MastNodeExt` trait into this one - and perhaps use that name instead? It's not exactly an `Ext` in the conventional sense, but probably still better than `MastNodeTrait`?

Reply:
See the PR description, section Technical details, second paragraph, and TODO. Thanks for the name suggestion!